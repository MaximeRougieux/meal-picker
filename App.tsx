/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the TypeScript template
 * https://github.com/react-native-community/react-native-template-typescript
 *
 * @format
 */

import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import React from 'react';
import { SafeAreaView, StyleProp, ViewStyle } from 'react-native';
import { RecipeDetails } from 'recipes/screens/RecipeDetails';
import { RecipeList } from 'recipes/screens/RecipeList';
import { RootStackParamsList } from 'utils/navigation/types';

const { Navigator, Screen } = createNativeStackNavigator<RootStackParamsList>();

const App = () => {
  const backgroundStyle: StyleProp<ViewStyle> = {
    backgroundColor: '#EAEAEA',
    flex: 1,
  };

  return (
    <SafeAreaView style={backgroundStyle}>
      <NavigationContainer>
        <Navigator initialRouteName="Home">
          <Screen
            name="Home"
            component={RecipeList}
            options={{ headerShown: false }}
          />
          <Screen
            name="Details"
            component={RecipeDetails}
            options={{
              headerTitle: '',
              headerStyle: { backgroundColor: '#EAEAEA' },
            }}
          />
        </Navigator>
      </NavigationContainer>
    </SafeAreaView>
  );
};

export default App;
