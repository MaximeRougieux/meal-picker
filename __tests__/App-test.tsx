/**
 * @format
 */

import React from 'react';
import App from '../App';
import nock from 'nock';

// Note: test renderer must be required after react-native.
import {
  render,
  waitForElementToBeRemoved,
} from '@testing-library/react-native';

it('renders correctly', async () => {
  nock.disableNetConnect();
  const { toJSON, queryByHintText } = render(<App />);
  await waitForElementToBeRemoved(() => queryByHintText('loading'));
  expect(toJSON()).toMatchSnapshot();
});
