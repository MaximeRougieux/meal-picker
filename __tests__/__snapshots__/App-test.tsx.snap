// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#F3F3F3",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 36,
        "marginHorizontal": 8,
      }
    }
  >
    Welcome to Meal Picker
  </Text>
  <RCTScrollView>
    <View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 16,
          }
        }
      >
        Nock: Disallowed net connect for "www.themealdb.com:443/api/json/v1/1/search.php?s=Chicken"
      </Text>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
